dashboard/signals.py

```diff
@@ -0,0 +1,30 @@
+"""
+Signals for user profiles
+"""
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from dashboard.models import CachedEnrollment, ProgramEnrollment
+
+
+@receiver(post_save, sender=CachedEnrollment, dispatch_uid="update_programenrollment")
+def create_programenrollment(sender, instance, created, **kwargs):  # pylint: disable=unused-argument
+    """
+    Signal handler to create Program enrollment when the CachedEnrollment table is updated
+    """
+    if instance.data is not None:
+        ProgramEnrollment.objects.get_or_create(
+            user=instance.user,
+            program=instance.course_run.course.program
+        )
+    else:
+        # check if there is any other entry with data != None
+        # if not delete the enrollment
+        if CachedEnrollment.objects.filter(
+            user=instance.user,
+            course_run__course__program=instance.course_run.course.program
+        ).exclude(data__isnull=True).count() == 0:
+            ProgramEnrollment.objects.filter(
+                user=instance.user,
+                program=instance.course_run.course.program
+            ).delete()
```

Comment:
a few things

1. this else statement seems problematic. we add a _lot_ of CachedEnrollment records with empty data, right? as I understand it, it's (total # of users) x (total # of course runs) - (actual user enrollments). this else clause would be triggered on the insertion of _every one_ of those, and the 'if' statement would evaluate to true for almost every one. my understanding is that we only want to delete the ProgramEnrollment if the CachedEnrollment had some data previously. could you use `if kwargs.get('created', False):` or something like that to make sure that this is an update rather than an insertion? 
1. i would prefer the `CachedEnrollment.objects.filter...` query to be stored in a local variable (`active_enrollment_count` or something) for readability
1. since `instance.user` and `instance.course_run.course.program` are used multiple times in this function, and in both branches of logic, i think it makes sense to store them in local variables